detox/src/utils/exec.js

```diff
@@ -164,22 +163,33 @@ function _joinCommandAndFlags(command, flags) {
   return result;
 }
 
-async function interruptProcess(childProcessPromise, signal = 'SIGINT') {
-  const log = execLogger.child({ fn: 'interruptProcess' });
+const DEFAULT_KILL_SCHEDULE = {
+  SIGINT: 0,
+};
 
+async function interruptProcess(childProcessPromise, schedule) {
   const childProcess = childProcessPromise.childProcess;
   const pid = childProcess.pid;
   const spawnargs = childProcess.spawnargs.join(' ');
+  const log = execLogger.child({ event: 'SPAWN_KILL', trackingId: pid });
 
-  log.debug({ event: 'EXEC_KILL', signal, process_pid: pid }, `sending ${signal} to [pid = ${pid}]: ${spawnargs}`);
-
-  childProcess.kill(signal);
-  await childProcessPromise.catch(e => {
-    /* istanbul ignore if */
-    if (e.exitCode != null) {
-      throw e;
-    }
+  const handles = _.mapValues({ ...DEFAULT_KILL_SCHEDULE, ...schedule }, (ms, signal) => {
```

Comment:
I wonder whether a provided schedule should in fact override the default... wdyt?